app/src/main/java/com/hedvig/app/feature/chat/data/UserRepository.kt

```diff
@@ -1,20 +1,24 @@
 package com.hedvig.app.feature.chat.data
 
-import com.apollographql.apollo.rx2.Rx2Apollo
 import com.hedvig.android.owldroid.graphql.AuthStatusSubscription
 import com.hedvig.android.owldroid.graphql.LogoutMutation
 import com.hedvig.android.owldroid.graphql.NewSessionMutation
 import com.hedvig.android.owldroid.graphql.SwedishBankIdAuthMutation
 import com.hedvig.app.ApolloClientWrapper
+import com.hedvig.app.util.apollo.toDeferred
+import com.hedvig.app.util.apollo.toFlow
 
 class UserRepository(private val apolloClientWrapper: ApolloClientWrapper) {
 
-    fun newUserSession() = Rx2Apollo.from(apolloClientWrapper.apolloClient.mutate(NewSessionMutation()))
+    suspend fun newUserSession() =
```

Comment:
We don't need to make this a `suspend`-function and `await` it, we can simply return the `Deferred`. Awaiting it makes assumptions about how the coroutine should be consumed that we do not need to make 🙂

Reply:
so should i call it like this `val response = runCatching { claimsRepository.fetchCommonClaimsAsync().await() }` in the viewModel instead?

Reply:
Correct!